gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -74,95 +87,128 @@ void parse(String config) throws IOException {
 
   private void parseConfig(Map<String, ?> config) {
     if (config != null) {
-      Boolean value = JsonUtil.getBoolean(config, "enable_cloud_logging");
-      if (value != null) {
-        enableCloudLogging = value;
-      }
-      value = JsonUtil.getBoolean(config, "enable_cloud_monitoring");
-      if (value != null) {
-        enableCloudMonitoring = value;
-      }
-      value = JsonUtil.getBoolean(config, "enable_cloud_trace");
-      if (value != null) {
-        enableCloudTracing = value;
-      }
-      destinationProjectId = JsonUtil.getString(config, "destination_project_id");
-      flushMessageCount = JsonUtil.getNumberAsLong(config, "flush_message_count");
-      List<?> rawList = JsonUtil.getList(config, "log_filters");
-      if (rawList != null) {
-        List<Map<String, ?>> jsonLogFilters = JsonUtil.checkObjectList(rawList);
-        ImmutableList.Builder<LogFilter> logFiltersBuilder = new ImmutableList.Builder<>();
-        for (Map<String, ?> jsonLogFilter : jsonLogFilters) {
-          logFiltersBuilder.add(parseJsonLogFilter(jsonLogFilter));
-        }
-        this.logFilters = logFiltersBuilder.build();
+      projectId = JsonUtil.getString(config, "project_id");
+      parseLoggingObject(config, CONFIG_LOGGING_VAR_NAME);
+      parseMonitoringObject(config, CONFIG_MONITORING_VAR_NAME);
+      parseTracingObject(config, CONFIG_TRACING_VAR_NAME);
+      parseCustomTags(config, CONFIG_LABELS_VAR_NAME);
+    }
+  }
+
+  private void parseLoggingObject(Map<String, ?> config,  String jsonObjectName) {
+    Map<String, ?> rawCloudLogging = JsonUtil.getObject(config, jsonObjectName);
+    if (rawCloudLogging != null) {
+      enableCloudLogging = true;
+      List<?> rawClientRpcEvents = JsonUtil.getList(rawCloudLogging, "client_rpc_events");
+      clientLogFilters =
+          rawClientRpcEvents != null ? parseRpcEvents(rawClientRpcEvents) : clientLogFilters;
```

Comment:
Can't this be simplified as simply
```
clientLogFilters = parseRpcEvents(rawClientRpcEvents);
```
`parseRpcEvents` returns null when argument is null. Also isn't the old value of `clientLogFilters` is null so why have `: clientLogFilters`?

Reply:
Updated `parseRpcEvents` to take `ImmutableList.Builder` as arguments to constructs filter lists.